datadog/fwprovider/data_source_datadog_azure_uc_config.go

```diff
@@ -0,0 +1,192 @@
+					},
+					"storage_container": schema.StringAttribute{
+						Computed:    true,
+						Description: "The name of the storage container where the Azure Export is saved.",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *datadogAzureUcConfigDataSource) Read(ctx context.Context, request datasource.ReadRequest, response *datasource.ReadResponse) {
+	var state datadogAzureUcConfigDataSourceModel
+	response.Diagnostics.Append(request.Config.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	cloudAccountId := state.CloudAccountId.ValueInt64()
+	ddResp, _, err := d.Api.GetCostAzureUCConfig(d.Auth, cloudAccountId)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error getting datadog Azure UC config"))
+		return
+	}
+
+	d.updateState(ctx, &state, &ddResp)
+
+	// Save data into Terraform state
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (d *datadogAzureUcConfigDataSource) updateState(ctx context.Context, state *datadogAzureUcConfigDataSourceModel, ucConfigPair *datadogV2.UCConfigPair) {
+	if data, ok := ucConfigPair.GetDataOk(); ok {
+		if id, ok := data.GetIdOk(); ok {
+			state.ID = types.StringValue(*id)
+		}
+
+		if attributes, ok := data.GetAttributesOk(); ok {
+			if configs, ok := attributes.GetConfigsOk(); ok && len(*configs) > 0 {
+				configs := *configs
+
+				// Set shared attributes from first config
+				firstConfig := configs[0]
+				state.AccountId = types.StringValue(firstConfig.GetAccountId())
+				state.ClientId = types.StringValue(firstConfig.GetClientId())
+				state.Scope = types.StringValue(firstConfig.GetScope())
+				state.Status = types.StringValue(firstConfig.GetStatus())
+				state.CreatedAt = types.StringValue(firstConfig.GetCreatedAt())
+
+				// Separate configs by dataset_type and populate respective blocks
+				for _, configData := range configs {
+					datasetType := configData.GetDatasetType()
+					if datasetType == "actual" {
+						state.ActualBillConfig = &actualBillConfigModel{}
+						state.ActualBillConfig.ExportName = types.StringValue(configData.GetExportName())
+						state.ActualBillConfig.ExportPath = types.StringValue(configData.GetExportPath())
+						state.ActualBillConfig.StorageAccount = types.StringValue(configData.GetStorageAccount())
+						state.ActualBillConfig.StorageContainer = types.StringValue(configData.GetStorageContainer())
+					} else if datasetType == "amortized" {
+						state.AmortizedBillConfig = &amortizedBillConfigModel{}
+						state.AmortizedBillConfig.ExportName = types.StringValue(configData.GetExportName())
+						state.AmortizedBillConfig.ExportPath = types.StringValue(configData.GetExportPath())
+						state.AmortizedBillConfig.StorageAccount = types.StringValue(configData.GetStorageAccount())
+						state.AmortizedBillConfig.StorageContainer = types.StringValue(configData.GetStorageContainer())
+					}
+				}
+			}
+		}
+	}
```

Comment:
Nit: dataset type can only be "actual" or "amortized" right? I'd maybe `switch` and have a default which is an error if unrecognized type.